README.md

```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
+
+The creation of Dynamic Instructions was the breakthrough that gave us the best of both worlds.  They can be authored in levelbuilder, use the existing text-to-speech system, and can also be used outside of AI Lab.  They encourage instructions to be written for just the current panel on the screen.  (They can also be triggered by other states in the app for specific instructions at that moment in time).  They do not dynamically resize, so the screen below them does not shift around, and they are guaranteed to always be fully readable without needing scrolling.  And best of all, they empower curriculum designers to explain what's happening in each level with minimal changes required in the app itself.  (That said, showing new Dynamic Instructions for specific states in the app is very easy.)
+
+#### Responsiveness
+The app is horizontally responsive, though it doesn't need to collapse all the way to mobile widths because our host site manages the viewport on mobile devices and also enforces landscape viewing.
+
+The app is vertically responsive, and uses flexbox to fill the vertical space with a combination of fixed- and variable-height elements.
+
+### Scenes:
+
+#### Select dataset
+This is usually the first scene, and can offer selection of both "pre-canned" datasets, which have accompanying metadata, or user-uploaded CSV.  The tiles use an art style somewhat consistent with that used elsewhere in our product.  There is a fun "grow" animation on tile hover, just to feel a little more interactive, which is reused for the A.I. bot head elsewhere in the app.
+
+#### Data display (label & features)
+This scene went through major iteration, before settling on the current implementation.  The scene is used twice, once to select a label and again to select features.  It tries to achieve a number of objectives:
+- it centers the tabular data in the student experience;
+- it allows the construction of a statement ("Predict [feature] based on [labels].") which carries across the experience;
+- it allows for the exploration of columns and their properties before they are selected.
+
+When selecting a label, the properties panel on the right shows information for the highlighted column:
+- a bar graph for categorical data;
+- min/max/range for numerical data.
+
+When selecting a feature, it shows the above properties for the highlighted column, but also attempts to show information about the relationship between that column and the previously-selected label:
+- a custom CrossTab when both label and feature are categorical data;
+- a scatterplot when both label and feature are numerical data.
+
+For student-uploaded CSV files, we attempt to guess the column type, but provide a way for the student to override this.
```

Comment:
maybe "we attempt to guess the column type based on the type of data detected in the column"?